src/dbnode/encoding/m3tsz/iterator.go

```diff
@@ -232,16 +144,20 @@ func (it *readerIterator) readFullFloatVal() {
 }
 
 func (it *readerIterator) readFloatXOR() {
-	it.xor = it.readXOR()
+	var err error
+	it.xor, err = ReadXOR(it.is, it.xor)
+	if err != nil {
+		it.err = err
+	}
 	it.vb ^= it.xor
 }
 
 func (it *readerIterator) readIntSigMult() {
 	if it.readBits(1) == opcodeUpdateSig {
-		if it.readBits(1) == opcodeZeroSig {
+		if it.readBits(1) == OpcodeZeroSig {
```

Comment:
Can't we unexport these and just use these internally and export an `IntIterator` or similar?  I assume not, but am curious to the backstory here.

Reply:
Still have to keep these exported since the int iterator is in the prot package now